core/src/main/scala/neotypes/Session.scala

```diff
@@ -6,28 +6,47 @@ import internal.syntax.stage._
 import org.neo4j.driver.{TransactionConfig => NeoTransactionConfig}
 import org.neo4j.driver.async.{AsyncSession => NeoAsyncSession}
 
-final class Session[F[_]] private[neotypes] (private val session: NeoAsyncSession) {
-  def transaction(implicit F: Async[F]): F[Transaction[F]] =
-    transaction(NeoTransactionConfig.empty)
-
-  def transaction(config: NeoTransactionConfig)(implicit F: Async[F]): F[Transaction[F]] =
-    F.async { cb =>
-      session.beginTransactionAsync(config).accept(cb) { tx =>
-        Right(new Transaction(tx))
+sealed trait Session[F[_]] {
+  def transaction: F[Transaction[F]]
+
+  def transaction(config: NeoTransactionConfig): F[Transaction[F]]
```

Comment:
I'm a bit curious about the shift to remove the implicit Async[F].  Does this represent a cleaner API?

Reply:
Yeah, basically the idea that users of **Session** shouldn't need the `Async` in scope, just the creator.
This also means that it may be possible to implement more **Sessions** that require different constraints.